### DISCUSSION TITLE
Adding more fields in checkout form

### DISCUSSION THREAD

**Discussion by boxofficeben** [woocommerce/woocommerce#45588] [Checkout Flow]
Will there be any updates where you are able to add more fields without having to pay for additional extensions or at least change the default fields, for example, if I didn't want to use the default company field and change the field name to something else, that would be helpful. The block looks so much nicer than the classic editor. Thanks!

**Comment by senadir** [comment]
Hey! From UI, it's not possible yet. If you're comfortable with code, you can follow [this tutorial to add extra fields in Checkout.](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/additional-checkout-fields.md)

You can disable the company field directly in the editor:

<img width="292" alt="image" src="https://github.com/woocommerce/woocommerce/assets/6165348/742fa42b-49a5-4f49-8fda-77f98af50ce3">


If you want to keep the company field but change its label, you can do this:
```php
add_filter(
	'woocommerce_get_country_locale_base',
	function ( $locale ) {
		$locale['company']['label'] = 'My custom label';
		return $locale;
	}
);
```

**Reply by ashishkerni** [reply]
i want to make phone number required in shipping but ever after checking the required phone number checkbox it is not working and showing phone number as optional field

**Reply by senadir** [reply]
Can you share a link to your website and possibly disable plugins to test this? It's been reported before that plugins causes conflict and prevent Checkout settings from being applied.

**Reply by ashishkerni** [reply]
Website :- https://Hayee.in
And there is no plugin installed for checkout modification

**Comment by jackmish** [comment]
June 2025 and still no convenient solution? I've only found method to switch to classic checkout form. Isnt it? 

New block checkout system without some fields would be usless in 95% of old one projects which i've made using another e-commerce platform.

Solution i've found: 1. Create new page, add "classic" checkout component 2. Kill infinite popup ".... turn new block system" 3. Change checkout view in WooCommerce->Settings->Advanced->checkout view
